### PR TITLE
Added testing for Stringtie

### LINKED ISSUE #105: Add tests for stringtie

### PR DESCRIPTION
**Pull request by sruthipsuresh**
Completed Stringtie testing with reverse/forward options. I've added an additional input file (a.gtf) - please let me know if taking the output of the gffread process (instead of having an input file already in the data folder) would be preferred in the workflow.
Closes #105 
@Emiller88

### CODE REVIEW

File: tests/software/stringtie/main.nf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { STRINGTIE as STRINGTIE_F } from '../../../software/stringtie/main.nf'  addParams( options: [ publish_dir:'test_stringtie_forward' ] )
```
Comment (drpatelh) [inline tests/software/stringtie/main.nf:5]:
```suggestion
include { STRINGTIE as STRINGTIE_FORWARD } from '../../../software/stringtie/main.nf'  addParams( options: [ publish_dir:'test_stringtie_forward' ] )
```

File: tests/software/stringtie/main.nf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+include { STRINGTIE as STRINGTIE_R } from '../../../software/stringtie/main.nf'  addParams( options: [ publish_dir:'test_stringtie_reverse' ] )
```
Comment (drpatelh) [inline tests/software/stringtie/main.nf:6]:
```suggestion
include { STRINGTIE as STRINGTIE_REVERSE } from '../../../software/stringtie/main.nf'  addParams( options: [ publish_dir:'test_stringtie_reverse' ] )
```

File: tests/software/stringtie/main.nf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+/*
+ * Test with forward strandedness
+ */
+workflow test_stringtie_forward {
+
+    def input = []
+    input = [ [ id:'test', strandedness:'forward' ], // meta map
+              [ file("${launchDir}/tests/data/bam/test.paired_end.sorted.bam", checkIfExists: true) ] ]
+
+    STRINGTIE_F (
```
Comment (drpatelh) [inline tests/software/stringtie/main.nf:17]: Indents are off here.

File: tests/software/stringtie/main.nf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+/*
+ * Test with reverse strandedness
+ */
+workflow test_stringtie_reverse {
+
+    def input = []
+    input = [ [ id:'test', strandedness:'reverse' ], // meta map
+              [ file("${launchDir}/tests/data/bam/test.paired_end.sorted.bam", checkIfExists: true) ] ]
+
+    STRINGTIE_R (
```
Comment (drpatelh) [inline tests/software/stringtie/main.nf:32]: Indents off

**Review by drpatelh — CHANGES_REQUESTED** [review]
Looks great! Some minor adjustments. I think we should keep the test data independent as it is now instead of chaining modules together in the tests. Just means the tests will pass quicker and will allow us to scale.

File: tests/software/stringtie/main.nf [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+/*
+ * Test with forward strandedness
+ */
+workflow test_stringtie_forward {
+    def input = []
+    input = [ [ id:'test', strandedness:'forward' ], // meta map
+              [ file("${launchDir}/tests/data/bam/test.paired_end.sorted.bam", checkIfExists: true) ] ]
+
+    STRINGTIE_FOWARD (
```
Comment (drpatelh) [inline tests/software/stringtie/main.nf:16]:
```suggestion
    STRINGTIE_FORWARD (
```

Comment (drpatelh) [reply tests/software/stringtie/main.nf:16]: Is that why the tests are failing?

**Review by drpatelh — APPROVED** [review]

File: tests/software/stringtie/main.nf (hunk 5 above)
Comment (Emiller88) [reply tests/software/stringtie/main.nf]: I'm gonna guess so :upside_down_face:

Comment (Emiller88) [reply tests/software/stringtie/main.nf]: We need to add to the CI that if the step fails then it uploads the logs so we investigate the error messages and not just guess.

**Pull request merged**